Source/WebCore/editing/cocoa/WebContentReaderCocoa.mm

```diff
@@ -419,6 +419,34 @@ static void replaceRichContentWithAttachments(LocalFrame& frame, DocumentFragmen
 #endif
 }
 
+static void simplifyFragmentForSingleTextAttachment(NSAttributedString *string, DocumentFragment& fragment)
+{
+    auto stringLength = string.length;
+    if (!stringLength || stringLength > 2)
+        return;
+
+    RetainPtr plainText = [string string];
+    if ([plainText characterAtIndex:0] != objectReplacementCharacter)
+        return;
+
+    __block unsigned numberOfTextAttachments = 0;
+    [string enumerateAttribute:NSAttachmentAttributeName inRange:NSMakeRange(0, 1) options:0 usingBlock:^(NSTextAttachment *attachment, NSRange, BOOL *) {
+        if (attachment)
+            numberOfTextAttachments++;
+    }];
+
+    if (numberOfTextAttachments != 1)
+        return;
+
+    if (stringLength == 2 && [plainText characterAtIndex:1] != newlineCharacter)
+        return;
+
+    if (RefPtr foundImage = descendantsOfType<HTMLImageElement>(fragment).first()) {
```

Comment:
I wonder if this will break pasting Genmoji...

Reply:
That's a good point — it should probably pull out the topmost `picture` or `img` element (whichever is first in preorder traversal).

That said, I think this would imply that Genmoji currently don't paste into Gmail though (since the picture element would cause the attachment to be dropped). Perhaps we need a separate quirk to hoist the inner `img` element out of the picture, just for Gmail compose...